### PR TITLE
refactor: enhance ComponentWorkflowRunStatusResponse with live obs flag

### PR DESCRIPTION
**Pull request by akila-i**
## Purpose
Refactor the previous `logUrl` field to now have `hasLiveObservability` field to be more extensible. This flag is then used to initialize openchoreoApiCLient or observabiliyApiClient to fetch logs/events accordingly instead of blindly using a provided logUrl.

## Approach
- Updated the ComponentWorkflowRunStatusResponse model to include HasLiveObservability field, indicating if the workflow run has live observability based on its age.
- Refactored GetComponentWorkflowRunStatus method to determine live observability without relying on a separate URL calculation.

## Checklist
- [ ] Tests added or updated (unit, integration, etc.)
- [ ] Samples updated (if applicable)

### Changed Files Breakdown by Folder
- **internal/**: 2 files changed
  - `openchoreo-api/models/`: 1 file (response.go)
  - `openchoreo-api/services/`: 1 file (component_workflow_service.go)

Lines changed: +11 / -53 (net -42)

---

### API / CRD Surface Changes
- ComponentWorkflowRunStatusResponse (JSON response for the workflow-run status endpoint)
  - Removed: `logUrl` (LogURL string)
  - Added: `hasLiveObservability` (HasLiveObservability bool)
- Affected endpoint: GET status for component workflow runs (service/handler).
- Compatibility risk: HIGH — this is a breaking change to a public API response schema with no backward compatibility layer or deprecation path.

---

### Implementation Details
- Removed construction/return of an observability LogURL.
- Added HasLiveObservability flag computed as age < TTL (TTL hardcoded to 1 hour).
- Service now decides whether a run has "live observability" based on run age; Status and Steps mapping preserved.

---

### Tests & Coverage
- Tests added/updated: none identified in this PR.
- Coverage gaps: no unit tests for the new HasLiveObservability logic; no handler/integration tests validating the changed JSON response; no edge-case tests around the TTL boundary (exactly 1 hour).

---

### Risk Hotspots
1. Breaking API change (High) — clients expecting `logUrl` will fail to deserialize or lose functionality; no versioning/deprecation.
2. Hardcoded TTL (Medium) — 1-hour magic number (TODO noted to source from cluster/workflow template).
3. Observability flow coupling (Medium) — flag introduced but downstream client switching behavior not visible here; potential integration gaps.
4. Tests missing (Medium) — lack of unit/integration coverage for behavior and schema change increases regression risk.

---

### Summary Statistics
- Files changed: 2 (internal/)
- Tests changed: 0
- API endpoints affected: 1 (component workflow run status)
- Field removals: 1 (LogURL)
- Field additions: 1 (HasLiveObservability)
- Lines added/removed: +11 / -53

### CODE REVIEW

File: internal/openchoreo-api/services/component_workflow_service.go [hunk 1]
```diff
@@ -524,63 +524,18 @@ func (s *ComponentWorkflowService) GetComponentWorkflowRunStatus(ctx context.Con
...
-		}
-		return "", fmt.Errorf("failed to get observability plane: %w", err)
-	}
-
-	return result.GetObserverURL(), nil
+	return &models.ComponentWorkflowRunStatusResponse{
+		Status:               overallStatus,
+		Steps:                steps,
+		HasLiveObservability: age < ttl,
+	}, nil
```
[the remark this answers is not in the document]
Comment (akila-i) [reply internal/openchoreo-api/services/component_workflow_service.go:538]: This will be fixed once the fetching TTL from workflowrun is implemented. Tracked through https://github.com/openchoreo/openchoreo/issues/1856

**Review by yashodgayashan — APPROVED** [review]

File: internal/openchoreo-api/services/component_workflow_service.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chalindukodikara) [reply internal/openchoreo-api/services/component_workflow_service.go:538]: We don't need to check TTL in WorkflowRun. WorkflowRun exists means workflowRun is not deleted. We just need to confirm that whether argo workflow exists in the build plane, and if it exists, we can retrieve logs. @akila-i
